### PR TITLE
GH-8156: move group title component to lib

### LINKED ISSUE #8156: Migrate prevNext, groupTitle, configMessage to new library

### PR DESCRIPTION
**Pull request by steinsebastian**
Migrate prevNext, groupTitle, configMessage to new library

### CODE REVIEW

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.spec.ts [hunk 1]
```diff
@@ -10,24 +10,25 @@ import {
   I18nTestingModule,
   RoutingService,
 } from '@spartacus/core';
+import { MessageConfig } from '@spartacus/storefront';
 import { Observable, of } from 'rxjs';
-import { MessageConfig } from '../config/message-config';
-import * as ConfigurationTestData from '../configuration-test-data';
+import * as ConfigurationTestData from '../../../../../../../projects/storefrontlib/src/cms-components/configurator/commons/configuration-test-data';
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.spec.ts]: I guess other tests are using this `configuration-test-data` which is preventing you to move it to the product library now?

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts [hunk 2]
```diff
@@ -60,7 +58,7 @@ export class ConfigPreviousNextButtonsComponent implements OnInit {
       );
   }
 
-  scrollToVariantConfigurationHeader() {
+  protected scrollToVariantConfigurationHeader() {
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: Please specify a return type.
Please do the same for all the methods.

File: feature-libs/product/configurators/common/src/core/common-configurator-core.module.ts [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import { NgModule } from '@angular/core';
 
-@NgModule({})
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/core/common-configurator-core.module.ts:3]: Nice catch!

File: projects/storefrontlib/src/cms-components/configurator/commons/config-form/config-form.component.spec.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ import {
 import { ICON_TYPE } from '@spartacus/storefront';
 import { cold } from 'jasmine-marbles';
 import { Observable, of } from 'rxjs';
+import { ConfigPreviousNextButtonsComponent } from '../../../../../../../feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component';
```
Comment (znikola) [inline projects/storefrontlib/src/cms-components/configurator/commons/config-form/config-form.component.spec.ts:19]: Is there a plan to move this test to the product library?

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  constructor(
+    protected configuratorCommonsService: ConfiguratorCommonsService,
+    protected configRouterExtractorService: ConfigRouterExtractorService,
+    protected config: MessageConfig
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription = this.hasPendingChanges$.subscribe(
+      (hasPendingChangesOrIsLoading) =>
+        this.showUpdateMessage(hasPendingChangesOrIsLoading.valueOf())
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: No need to use `valueOf()`, if the return type is small `boolean`.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+          switchMap((hasPendingChanges) =>
+            this.configuratorCommonsService
+              .isConfigurationLoading(routerData.owner)
+              .pipe(map((isLoading) => hasPendingChanges || isLoading))
+          )
+        );
+    })
+  );
+
+  protected subscription: Subscription;
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: You can instantiate `subscription` here, inline. 
To use it, you can just do `this.subscription.add()` in `ngOnInit`.
By doing this, you can avoid the `if (this.subscription)` in `ngOnDestroy`.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  protected showUpdateMessage(hasPendingChangesOrIsLoading: boolean): void {
+    const updateMessageElement = document.getElementById(
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: Let's avoid using `document` directly, as it's not available in SSR app and will cause an error.
This applies to all the cases when doing the DOM manipulations directly, such as `updateMessageElement.classList`

For these kind of cases it's better to use the platform agnostic `Renderer2` from Angular.
However, before refactoring I'd like to take a look at this code more closely.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      updateMessageElementId
+    );
+
+    if (hasPendingChangesOrIsLoading) {
+      this.changesInProgress = true;
+
+      setTimeout(() => {
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: I'm curious why was this necessary? Did you have any errors thrown by Angular (e.g. `ExpressionChangedAfterItHasBeenCheckedError`)?
Or was it used because you have a business requirement to take the `this.config?.updateConfigurationMessage?.waitingTime` into an account? If this is the case, we can come up with a more reactive approach

File: feature-libs/product/configurators/variant/src/interactive-configuration.module.ts [hunk 9]
```diff
@@ -3,18 +3,18 @@ import { RouterModule } from '@angular/router';
 import { CmsConfig, ConfigModule, ProductModule } from '@spartacus/core';
 import {
   CommonConfiguratorComponentsModule,
+  ConfigGroupTitleComponent,
```
Comment (znikola) [inline feature-libs/product/configurators/variant/src/interactive-configuration.module.ts]: Let's try to clean up this file a bit, as it feels a bit overcrowded :)
For example, this mapping part that's located in the configuration on line 49:
```ts
 VariantConfigurationGroupTitle: {
          component: ConfigGroupTitleComponent,
        },
        },
```
can be done in the module where `ConfigGroupTitleComponent` is registered.
By doing it, we can just import the component modules to this module.

Comment (znikola) [reply feature-libs/product/configurators/variant/src/interactive-configuration.module.ts]: Let's start simple, and do it only for the three newly imported components.
But going forward, I'd love to see this principle applied to other components in this module.

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts [hunk 10]
```diff
@@ -60,7 +58,7 @@ export class ConfigPreviousNextButtonsComponent implements OnInit {
...
     const theElement = document.querySelector('.VariantConfigurationTemplate');
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: As stated previously, we should avoid using the `document` directly, as it's not available in SSR mode.
I'd also like to take a closer look at these kind of DOM manipulations, and check if there's an alternative way.

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts [hunk 11]
```diff
@@ -60,7 +58,7 @@ export class ConfigPreviousNextButtonsComponent implements OnInit {
       );
   }
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts:60]: On line 70 and 76, please use small `boolean` instead of `Boolean`. 
The big `Boolean` is a constructor, while the small one is a type.

Comment (znikola) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts:60]: Can you also please check other places for this? I think searching for `<Boolean>` is a good enough search that'll yield the same mistake.
Because of prettier formatting, you should also try this search:
```ts
<
    Boolean
  >
```

I think there's one case in `ConfigMessageComponent`

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+    }
+  }
+
+  protected showUpdateMessage(hasPendingChangesOrIsLoading: boolean): void {
+    const updateMessageElement = document.getElementById(
+      updateMessageElementId
+    );
+
+    if (hasPendingChangesOrIsLoading) {
+      this.changesInProgress = true;
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: Performing these kind of side effects withing the stream (setting and accessing a private class property, using `setTimeout`) is not considered as a good practice. 
Is there an alternative approach that we can take?

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+} from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+
+const updateMessageElementId = 'cx-config-update-message';
+
+@Component({
+  selector: 'cx-config-message',
+  templateUrl: './config-message.component.html',
+  encapsulation: ViewEncapsulation.None,
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: I'm curious, why was this necessary?

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 14]
```diff
@@ -0,0 +1,75 @@
+import { Component, OnDestroy, OnInit, ViewEncapsulation } from '@angular/core';
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: I tried to quickly look into this class, and this is something I came up with. Please bear in mind that:
- is the waitingTime from `this.config?.updateConfigurationMessage?.waitingTime` a requirement that has to be met? If that's the case, this solution of mine is not complete.
- I might've missunderstood the idea behind this code. If so, let's discuss it
- I haven't tested it, and it might not work

That being said, here's the TS file:
```ts
import { Component } from '@angular/core';
import { ConfiguratorCommonsService } from '@spartacus/core';
import { ConfigRouterExtractorService } from '@spartacus/storefront';
import { Observable } from 'rxjs';
import { map, switchMap } from 'rxjs/operators';

@Component({
  selector: 'cx-config-message',
  templateUrl: './config-message.component.html',
})
export class ConfigMessageComponent {
  protected hasPendingChanges$: Observable<
    Boolean
  > = this.configRouterExtractorService
    .extractRouterData()
    .pipe(
      switchMap((routerData) =>
        this.configuratorCommonsService
          .hasPendingChanges(routerData.owner)
          .pipe(
            switchMap((hasPendingChanges) =>
              this.configuratorCommonsService
                .isConfigurationLoading(routerData.owner)
                .pipe(map((isLoading) => hasPendingChanges || isLoading))
            )
          )
      )
    );

  visibleClass$: Observable<string> = this.hasPendingChanges$.pipe(
    map((hasPendingChangesOrIsLoading) =>
      hasPendingChangesOrIsLoading ? 'd-none' : ''
    )
  );

  constructor(
    protected configuratorCommonsService: ConfiguratorCommonsService,
    protected configRouterExtractorService: ConfigRouterExtractorService
  ) {}
}

```

and here's the html template:
```html
<div class="wrapper">
  <diV id="cx-config-update-message" [class]="visibleClass$ | async">
    <cx-spinner></cx-spinner>
    <strong>{{ 'configurator.header.updateMessage' | cxTranslate }}</strong>
  </diV>
</div>

```

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts [hunk 15]
```diff
@@ -1,38 +1,36 @@
...
-  ngOnInit(): void {
-    this.configuration$ = this.configRouterExtractorService
-      .extractRouterData()
-      .pipe(
-        switchMap((routerData) =>
-          this.configuratorCommonsService.getConfiguration(routerData.owner)
-        )
-      );
-  }
-
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: At very least we can get rid of either `onPrevious` or `navigateToPreviousGroup` methods, and just have one that will be called on the click event. 
Similar can be done for the `onNext`.

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts (hunk 10 above)
Comment (znikola) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: Can you help me understand what is this component doing?
And can you tell me how to see it in action?

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.spec.ts (hunk 1 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.spec.ts]: Correct.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 6 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: ok

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 5 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: ok

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts (hunk 15 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: ok

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts (hunk 11 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts:60]: ok

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts (hunk 2 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: ok

File: feature-libs/product/configurators/common/src/core/common-configurator-core.module.ts (hunk 3 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/core/common-configurator-core.module.ts:3]: ok

File: projects/storefrontlib/src/cms-components/configurator/commons/config-form/config-form.component.spec.ts (hunk 4 above)
Comment (steinsebastian) [reply projects/storefrontlib/src/cms-components/configurator/commons/config-form/config-form.component.spec.ts:19]: Yes, Christoph is working on this.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 14 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: I think we need to discuss the whole component with regards to setting a timeout and the DOM manipulation.
As I only moved the component and I was not the original developer I'd like to involve my colleague Lukas (@schuppell) here.

Let's have a call next week regarding this.
As an upfront information: this component shows some kind of loading message ("Configuration is updated wait a little bit!") if the calculation of the configuration-state in the configuration-engine in the backend takes a little bit longer (e.g. due to complexity of calculation or low bandwidth).
In order to avoid that this message is constantly flickering although the response time of the engine is good, we decided to show the message only after a certain time if the calculation is still not finished. Thus the timeout.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 13 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: Me too. ;-) Let's discuss in our session.

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts (hunk 10 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: I can show you this in our session next week, but here is already the explanation:
The list of attributes can be quite long, so the user scrolls down and at the end there are the previous/next buttons. If he clicks on the buttons the application navigates to the previous/next group of attributes. Without the method and if the previous/next group is also quite long the user would end up somewhere in the middle or at the end of this group. Our UI designer decided that with every group change the UI should scroll to the start of the list of attributes so that the user is not missing attributes by accident.
Furthermore he decided that the UI should not scroll completely to the top but really to the start of the configuration area so that the user can concentrate on his work in the configuration session.

File: feature-libs/product/configurators/variant/src/interactive-configuration.module.ts (hunk 9 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/variant/src/interactive-configuration.module.ts]: Could you elaborate on this? I've tried several things but always ran into errors.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 14 above)
Comment (znikola) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: Ok, sounds good. If you want to add some delay before showing the message, there are rxjs operators that we can consider for this case.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 16]
```diff
@@ -16,8 +16,9 @@ const updateMessageElementId = 'cx-config-update-message';
 })
 export class ConfigMessageComponent implements OnInit, OnDestroy {
   changesInProgress = false;
+  subscription: Subscription = new Subscription();
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: Every property that you're not using in the template should be marked as `protected` (or `private` in special cases).

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 16 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: Done

File: feature-libs/product/configurators/variant/src/interactive-configuration.module.ts (hunk 9 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/variant/src/interactive-configuration.module.ts]: Done.

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts (hunk 10 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: Done. Disabled code execution for SSR.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 12 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: We have changed the config-message.component according to your latest suggestions. Please review.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 14 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: We have changed the config-message.component according to your latest suggestions. Please review.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.spec.ts (hunk 1 above)
Comment (znikola) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.spec.ts]: Let's do `import * as ConfigurationTestData from 'projects/storefrontlib...` then.
Let's also create a list of all the spec files that should be updated once the `ConfigurationTestData` is moved to the lib. We can use the `todo.md` file (located in the root of the project) for the list.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.html [hunk 17]
```diff
@@ -1,5 +1,5 @@
 <div class="wrapper">
-  <diV id="cx-config-update-message" class="d-none">
+  <diV id="cx-config-update-message" [class]="visibleClass$ | async">
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.html]: Ooops, `diV` instead of `div` :)

Comment (znikola) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.html]: The same is for the closing tag

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 18]
```diff
@@ -1,74 +1,40 @@
...
-  > = this.configRouterExtractorService.extractRouterData().pipe(
-    switchMap((routerData) => {
-      return this.configuratorCommonsService
-        .hasPendingChanges(routerData.owner)
-        .pipe(
-          switchMap((hasPendingChanges) =>
-            this.configuratorCommonsService
-              .isConfigurationLoading(routerData.owner)
-              .pipe(map((isLoading) => hasPendingChanges || isLoading))
+    Boolean
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: Please use lowercase `boolean`. 
`Boolean` is a ctor, whereas `boolean` is a type

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts [hunk 19]
```diff
@@ -27,9 +27,10 @@ export class ConfigMessageComponent {
       )
     );
 
-  visibleClass$: Observable<string> = this.hasPendingChanges$.pipe(
+  visibleClass$: Observable<string> = interval(1000).pipe(
+    withLatestFrom(this.hasPendingChanges$),
     map((hasPendingChangesOrIsLoading) =>
-      hasPendingChangesOrIsLoading ? 'visible' : ''
+      hasPendingChangesOrIsLoading[1] ? 'visible' : ''
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: Let's be more explicit, and specify that `map` receives an array:
```ts
map(([_, hasPendingChangesOrIsLoading]) => hasPendingChangesOrIsLoading ? 'visible' : '')
```

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts [hunk 20]
```diff
@@ -51,12 +58,17 @@ export class ConfigPreviousNextButtonsComponent {
   }
 
   protected scrollToVariantConfigurationHeader(): void {
-    const theElement = document.querySelector('.VariantConfigurationTemplate');
-    let topOffset = 0;
-    if (theElement instanceof HTMLElement) {
-      topOffset = theElement.offsetTop;
+    if (isPlatformBrowser(this.platformId)) {
+      // we don't want to run this logic when doing SSR
+      const theElement = document.querySelector(
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: njecting the platform is a way to go 👍

~~IBut maybe we can actually inject `protected windowRef: WindowRef` instead, and try the following?~~

```ts
  protected scrollToVariantConfigurationHeader(): void {
    const theElement = this.renderer.selectRootElement(
      '.VariantConfigurationTemplate'
    );
    let topOffset = 0;
    if (theElement instanceof HTMLElement) {
      topOffset = theElement.offsetTop;
    }
    this.windowRef.nativeWindow?.scroll(0, topOffset);
  }
```

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.spec.ts (hunk 1 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.spec.ts]: Done.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.html (hunk 17 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.html]: done.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 18 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: done.

File: feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts (hunk 19 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-message/config-message.component.ts]: done.

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts (hunk 20 above)
Comment (znikola) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: Actually, scratch the above.
Let's extract the logic as is, into a service which can be re-used by other components.
Please make the class name (`.VariantConfigurationTemplate`) in the service override-able. And the service should be smart enough to know whether to actually do the scrolling, based on the platform used.

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts (hunk 20 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: Just for clarification: should I extract the logic "as is" meaning as it was before or with the usage of the renderer?

Comment (znikola) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: Yes, before the renderer, and wrapped around the platform check

Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: I have moved it to config-utils.service.ts as "scrollToConfigurationElement(selectors: string): void ". 
Currently config-utils.service.ts is located in storefrontlib but will be moved to the lib later. I've added a todo for this in todo.md.

File: todo.md [hunk 21]
```diff
@@ -23,3 +23,5 @@
     - config-message.component.spec.ts
     - config-group-title.component.spec.ts
     - config-previous-next-buttons.component.spec.ts
+
+7. Update import of ConfigUtilsService in config-previous-next-buttons.components.ts once ConfigUtilsService has been moved to lib.
```
Comment (znikola) [inline todo.md]: I think you don't need to mention this one. When you move the from storefrontlib to the product lib, you'll get compilation errors, which can serve as a reminder that you need to fix imports.

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts [hunk 22]
```diff
@@ -35,11 +37,14 @@ export class ConfigPreviousNextButtonsComponent {
     protected configuratorGroupsService: ConfiguratorGroupsService,
     protected configuratorCommonsService: ConfiguratorCommonsService,
     protected configRouterExtractorService: ConfigRouterExtractorService,
+    protected configUtils: ConfigUtilsService,
     @Inject(PLATFORM_ID) protected platformId: any
```
Comment (znikola) [inline feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: You can remove this one now

File: projects/storefrontlib/src/cms-components/configurator/commons/service/config-utils.service.ts [hunk 23]
```diff
@@ -10,7 +11,10 @@ import { map, take } from 'rxjs/operators';
   providedIn: 'root',
 })
 export class ConfigUtilsService {
```
Comment (znikola) [inline projects/storefrontlib/src/cms-components/configurator/commons/service/config-utils.service.ts:13]: This generic config util service might be a good fit for this kind of method.
Do you anticipate some other UI-related util methods in the future? If yes, maybe we can create a dedicated service in that case.

**Review by znikola — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts (hunk 22 above)
Comment (steinsebastian) [reply feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts]: done.

File: projects/storefrontlib/src/cms-components/configurator/commons/service/config-utils.service.ts (hunk 23 above)
Comment (steinsebastian) [reply projects/storefrontlib/src/cms-components/configurator/commons/service/config-utils.service.ts:13]: I can't judge at the moment whether there will be other UI related stuff.

File: todo.md (hunk 21 above)
Comment (steinsebastian) [reply todo.md]: Ok. I'll remove it.

**Review by znikola — APPROVED** [review]
